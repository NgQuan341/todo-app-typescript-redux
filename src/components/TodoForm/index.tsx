import React, { useState } from 'react'
import { Col, Input, Button } from 'antd'
import { useDispatch } from 'react-redux'
import { v4 as uuid } from 'uuid'
// import { addTodo } from '../../redux/actions'
import todoListSlice, { addNewTodo } from '../TodoList/reducer'

const TodoForm = () => {
    const [todoName, setTodoName] = useState('')
    const dispatch = useDispatch();
    const handleAddButtonClick = () => {
        // dispatch(
        //     todoListSlice.actions.addTodo({
        //     id: uuid(),
        //     name: todoName,
        //     completed: false
        // }))
        dispatch<any>(addNewTodo(
            {
                id: uuid(),
                name: todoName,
                completed: false
            }
        ))
        setTodoName('')
    }
    const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
        setTodoName(e.target.value)
    }
    return (
        <Col span={24}>
            <Input.Group style={{ display: 'flex', marginTop: '20px' }} compact>
                <Input value={todoName} name='input-todo' onChange={handleInputChange} />
                <Button type='primary' name='submit-button' className='button-submit' style={{ background: '#2DEB56', borderColor: '#2DEB56', color: 'white' }} onClick={handleAddButtonClick}>
                    Add
                </Button>
            </Input.Group>
        </Col>
    )
}

export default TodoForm