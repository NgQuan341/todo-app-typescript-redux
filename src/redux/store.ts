// import {createStore} from 'redux'
// import rootReducer from './reducer';
// import { composeWithDevTools } from 'redux-devtools-extension'

// const composedEnheancers = composeWithDevTools()
// const store = createStore(rootReducer, composedEnheancers);

// export default store

import { configureStore } from '@reduxjs/toolkit'
import todoListSlice from '../components/TodoList/reducer';

const store = configureStore({
    reducer: todoListSlice.reducer
})
export type RootState = ReturnType<typeof store.getState>

export default store