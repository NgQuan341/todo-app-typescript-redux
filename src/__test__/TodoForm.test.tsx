import { shallow, mount } from 'enzyme'
import { Provider } from 'react-redux';
import TodoForm from '../components/TodoForm';
import store from '../redux/store';

const setupShallow = () => {
   let wrapper = shallow<typeof TodoForm>(<Provider store={store}><TodoForm /></Provider>)
   return wrapper
}
const setupMount = () => {
    let wrapper = mount<typeof TodoForm>(<Provider store={store}><TodoForm /></Provider>)
    return wrapper
 }
 
describe('TodoForm component', () =>{
    test('renders without error', () => {
        const wrapper = setupShallow()                 
        expect(wrapper.html()).toMatchSnapshot()
    })
    // test('input should call handleInputChange on change event', () =>{
    //     const wrapper = setupMount();
    //     const myMethod = jest.spyOn(wrapper,handleAddButtonClick)
    //     const event = {target:{value: 'todo1'}};

    // })
    // test ('test find button in TodoForm', () =>{
    //     const wrapper = setupMount()   
    //     expect(wrapper.find(`[name='submit-button']`)).toBeTruthy();            
    // })
})