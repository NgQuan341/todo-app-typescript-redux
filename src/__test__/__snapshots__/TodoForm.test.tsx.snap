// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoForm component renders without error 1`] = `"<div class=\\"ant-col ant-col-24\\"><span class=\\"ant-input-group ant-input-group-compact\\" style=\\"display:flex;margin-top:20px\\"><input type=\\"text\\" value=\\"\\" name=\\"input-todo\\" class=\\"ant-input\\"/><button name=\\"submit-button\\" style=\\"background:#2DEB56;border-color:#2DEB56;color:white\\" type=\\"button\\" class=\\"ant-btn ant-btn-primary button-submit\\"><span>Add</span></button></span></div>"`;
